dockstore-webservice/src/main/java/io/dockstore/webservice/resources/HostedWorkflowResource.java

```diff
@@ -112,10 +115,23 @@ protected Workflow getEntry(User user, String registry, String name, String desc
         workflow.setSourceControl(SourceControl.DOCKSTORE);
         workflow.setDescriptorType(descriptorType);
         workflow.setLastUpdated(new Date());
+        workflow.setDefaultWorkflowPath(getDefaultWorkflowPath(descriptorType));
         workflow.getUsers().add(user);
         return workflow;
     }
 
+    private String getDefaultWorkflowPath(String descriptorType) {
+        descriptorType = descriptorType.toLowerCase();
+        if (Objects.equals(descriptorType, "cwl")) {
```

Comment:
This relationship would be more clear by creating a map between descriptorType and default paths